sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuiteBase.scala

```diff
@@ -828,7 +828,7 @@ abstract class CastSuiteBase extends SparkFunSuite with ExpressionEvalHelper {
         val ret2 = cast(
           Literal.create(Map("1" -> "a".getBytes, "2" -> null, "3" -> "c".getBytes)),
           StringType)
-        checkEvaluation(ret2, s"${lb}1 -> a, 2 ->${if (legacyCast) "" else " null"}, 3 -> c$rb")
+        checkEvaluation(ret2, s"${lb}1 -> a, 2 -> ${if (legacyCast) "" else "null"}, 3 -> c$rb")
```

Comment:
can we avoid changing the cast behavior?

Reply:
I think it's a reasonable change, MapToString has three elements to print, key, value, and separator. that is `k separator v`, this approach is more unified. For Cast, the separator is `" -> "`, and for ToPrettyString, it is `":"`.

This affects only one behavior of the Cast, that is, when v of the first element is an empty string, it was printed as `k ->` before and now is `k -> `.  I think `k -> ` is more reasonable, it is consistent with the other elements except the first element.

For example, map(k1,"",k2,v2), before it's `k1 ->, k2 -> v2`, now it's `k1 -> , k2 -> v2`.

Reply:
looks reasonable. @sadikovi what do you think?